lib/spack/llnl/util/tty/__init__.py

```diff
@@ -141,6 +144,11 @@ def msg(message, *args, **kwargs):
     if not msg_enabled():
         return
 
+    if _osascript:
+        subprocess.call(['osascript', '-e', 'if app "Terminal" is frontmost '
```

Comment:
`osascript` might not be the same as `/usr/bin/osascript`

Reply:
That's fine, if someone has a newer version of `osascript` in their `PATH` then I don't see why we shouldn't use it. But the location on macOS should be pretty stable. I couldn't think of a simpler way to detect whether or not we are on macOS since I can't import most of the Spack libs.

Reply:
The last commit uses `which`, so `osascript` should be the same now.